pkg/install/resources.go

```diff
@@ -132,3 +139,81 @@ func VolumeSnapshotLocation(namespace, provider string, config map[string]string
 		},
 	}
 }
+
+func Secret(namespace string, data []byte) *corev1.Secret {
+	return &corev1.Secret{
+		ObjectMeta: objectMeta(namespace, "cloud-credentials"),
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: corev1.SchemeGroupVersion.String(),
+		},
+		Data: map[string][]byte{
+			"cloud": data,
+		},
+		Type: corev1.SecretTypeOpaque,
+	}
+}
+
+func appendUnstructured(list *unstructured.UnstructuredList, obj runtime.Object) error {
+	u, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&obj)
+	if err != nil {
+		return err
+	}
+	list.Items = append(list.Items, unstructured.Unstructured{Object: u})
+	return nil
+}
+
+// AllResources returns a list of all resources necessary to install Velero, in the appropriate order, into a Kubernetes cluster.
+// Items are unstructured, since there are different data types returned.
+func AllResources(namespace, image, backupStorageProviderName, bucketName, prefix string, bslConfig, vslConfig map[string]string, secretData []byte, restoreOnly bool) (*unstructured.UnstructuredList, error) {
```

Comment:
I would probably create an `Options` struct with the params, since the list is pretty long.